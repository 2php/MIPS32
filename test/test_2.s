#sra_sllv_srlv_lbu_sb
start:
  .text 0
  lw $2, operand($0)  # r2 := operand
  sra $3, $2, 3       # r3 := r2 >> 3 (arithmetic)
  sllv $2, $3, $2     # r2 := r3 << r2(4 downto 0, = 5)
  sw $2, operand($0)  # operand := r2
  lbu $2, shamt($0)   # r2 := shamt(7 downto 0, = 0x8E)
  lw $3, operand($0)  # r3 := operand
  srlv $3, $3, $2     # r3 := r3 >> r2(4 downto 0, = 14)
  addiu $2, $0, 1     # r2 := 1  
  sb $3, shamt($2)    # shamt(15 downto 8) := r3
  lw $2, shamt($0)    # r2 := shamt
  syscall
operand:
  .word 0xAA001005
shamt:
  .word 0xFFFFFF8E
  