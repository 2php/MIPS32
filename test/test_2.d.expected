booting
loading memory from file
80000000 3c028000
80000004 8c420048
80000008 000218c3
8000000c 00431004
80000010 3c018000
80000014 ac220048
80000018 3c028000
8000001c 9042004c
80000020 3c038000
80000024 8c630048
80000028 00431806
8000002c 24020001
80000030 3c018000
80000034 00220821
80000038 a023004c
8000003c 3c028000
80000040 8c42004c
80000044 0000000c
80000048 aa001005
8000004c ffffff8e

fetch instr @ 0x80000000 (10000000000000000000000000000000)
MMU[0x80000000] :  0x00000000
Mem[0x00000000] :  0x3c028000 (00111100000000101000000000000000)
001111 | 00000 | 00010 | 1000000000000000
LUI      rs: 0   rt: 2   imm:0x8000
R[ 2] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x80000004 (10000000000000000000000000000100)
MMU[0x80000004] :  0x00000004
Mem[0x00000004] :  0x8c420048 (10001100010000100000000001001000)
100011 | 00010 | 00010 | 0000000001001000
LW       rs: 2   rt: 2   imm:0x0048
R[ 2] :  0x80000000 (10000000000000000000000000000000)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x00000048 (00000000000000000000000001001000)
alu_r :  0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] :  0xaa001005 (10101010000000000001000000000101)
R[ 2] <= 0xaa001005 (10101010000000000001000000000101)

fetch instr @ 0x80000008 (10000000000000000000000000001000)
MMU[0x80000008] :  0x00000008
Mem[0x00000008] :  0x000218c3 (00000000000000100001100011000011)
000000 | 00000 | 00010 | 00011 | 00011 | 000011
SPECIAL  rs: 0   rt: 2   rd: 3   sa: 3   SRA    
R[ 2] :  0xaa001005 (10101010000000000001000000000101)
alu_a <= 0x00000003 (00000000000000000000000000000011)
alu_b <= 0xaa001005 (10101010000000000001000000000101)
alu_r :  0xf5400200 (11110101010000000000001000000000)
R[ 3] <= 0xf5400200 (11110101010000000000001000000000)

fetch instr @ 0x8000000c (10000000000000000000000000001100)
MMU[0x8000000c] :  0x0000000c
Mem[0x0000000c] :  0x00431004 (00000000010000110001000000000100)
000000 | 00010 | 00011 | 00010 | 00000 | 000100
SPECIAL  rs: 2   rt: 3   rd: 2   sa: 0   SLLV   
R[ 2] :  0xaa001005 (10101010000000000001000000000101)
R[ 3] :  0xf5400200 (11110101010000000000001000000000)
alu_a <= 0xaa001005 (10101010000000000001000000000101)
alu_b <= 0xf5400200 (11110101010000000000001000000000)
alu_r :  0xa8004000 (10101000000000000100000000000000)
R[ 2] <= 0xa8004000 (10101000000000000100000000000000)

fetch instr @ 0x80000010 (10000000000000000000000000010000)
MMU[0x80000010] :  0x00000010
Mem[0x00000010] :  0x3c018000 (00111100000000011000000000000000)
001111 | 00000 | 00001 | 1000000000000000
LUI      rs: 0   rt: 1   imm:0x8000
R[ 1] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x80000014 (10000000000000000000000000010100)
MMU[0x80000014] :  0x00000014
Mem[0x00000014] :  0xac220048 (10101100001000100000000001001000)
101011 | 00001 | 00010 | 0000000001001000
SW       rs: 1   rt: 2   imm:0x0048
R[ 1] :  0x80000000 (10000000000000000000000000000000)
R[ 2] :  0xa8004000 (10101000000000000100000000000000)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x00000048 (00000000000000000000000001001000)
alu_r :  0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] <= 0xa8004000 (10101000000000000100000000000000)

fetch instr @ 0x80000018 (10000000000000000000000000011000)
MMU[0x80000018] :  0x00000018
Mem[0x00000018] :  0x3c028000 (00111100000000101000000000000000)
001111 | 00000 | 00010 | 1000000000000000
LUI      rs: 0   rt: 2   imm:0x8000
R[ 2] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x8000001c (10000000000000000000000000011100)
MMU[0x8000001c] :  0x0000001c
Mem[0x0000001c] :  0x9042004c (10010000010000100000000001001100)
100100 | 00010 | 00010 | 0000000001001100
LBU      rs: 2   rt: 2   imm:0x004c
R[ 2] :  0x80000000 (10000000000000000000000000000000)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x0000004c (00000000000000000000000001001100)
alu_r :  0x8000004c (10000000000000000000000001001100)
MMU[0x8000004c] :  0x0000004c
Mem[0x0000004c] :  0x8e (10001110)
R[ 2] <= 0x0000008e (00000000000000000000000010001110)

fetch instr @ 0x80000020 (10000000000000000000000000100000)
MMU[0x80000020] :  0x00000020
Mem[0x00000020] :  0x3c038000 (00111100000000111000000000000000)
001111 | 00000 | 00011 | 1000000000000000
LUI      rs: 0   rt: 3   imm:0x8000
R[ 3] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x80000024 (10000000000000000000000000100100)
MMU[0x80000024] :  0x00000024
Mem[0x00000024] :  0x8c630048 (10001100011000110000000001001000)
100011 | 00011 | 00011 | 0000000001001000
LW       rs: 3   rt: 3   imm:0x0048
R[ 3] :  0x80000000 (10000000000000000000000000000000)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x00000048 (00000000000000000000000001001000)
alu_r :  0x80000048 (10000000000000000000000001001000)
MMU[0x80000048] :  0x00000048
Mem[0x00000048] :  0xa8004000 (10101000000000000100000000000000)
R[ 3] <= 0xa8004000 (10101000000000000100000000000000)

fetch instr @ 0x80000028 (10000000000000000000000000101000)
MMU[0x80000028] :  0x00000028
Mem[0x00000028] :  0x00431806 (00000000010000110001100000000110)
000000 | 00010 | 00011 | 00011 | 00000 | 000110
SPECIAL  rs: 2   rt: 3   rd: 3   sa: 0   SRLV   
R[ 2] :  0x0000008e (00000000000000000000000010001110)
R[ 3] :  0xa8004000 (10101000000000000100000000000000)
alu_a <= 0x0000008e (00000000000000000000000010001110)
alu_b <= 0xa8004000 (10101000000000000100000000000000)
alu_r :  0x0002a001 (00000000000000101010000000000001)
R[ 3] <= 0x0002a001 (00000000000000101010000000000001)

fetch instr @ 0x8000002c (10000000000000000000000000101100)
MMU[0x8000002c] :  0x0000002c
Mem[0x0000002c] :  0x24020001 (00100100000000100000000000000001)
001001 | 00000 | 00010 | 0000000000000001
ADDIU    rs: 0   rt: 2   imm:0x0001
R[ 0] :  0x00000000 (00000000000000000000000000000000)
alu_a <= 0x00000000 (00000000000000000000000000000000)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x00000001 (00000000000000000000000000000001)
R[ 2] <= 0x00000001 (00000000000000000000000000000001)

fetch instr @ 0x80000030 (10000000000000000000000000110000)
MMU[0x80000030] :  0x00000030
Mem[0x00000030] :  0x3c018000 (00111100000000011000000000000000)
001111 | 00000 | 00001 | 1000000000000000
LUI      rs: 0   rt: 1   imm:0x8000
R[ 1] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x80000034 (10000000000000000000000000110100)
MMU[0x80000034] :  0x00000034
Mem[0x00000034] :  0x00220821 (00000000001000100000100000100001)
000000 | 00001 | 00010 | 00001 | 00000 | 100001
SPECIAL  rs: 1   rt: 2   rd: 1   sa: 0   ADDU   
R[ 1] :  0x80000000 (10000000000000000000000000000000)
R[ 2] :  0x00000001 (00000000000000000000000000000001)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x00000001 (00000000000000000000000000000001)
alu_r :  0x80000001 (10000000000000000000000000000001)
R[ 1] <= 0x80000001 (10000000000000000000000000000001)

fetch instr @ 0x80000038 (10000000000000000000000000111000)
MMU[0x80000038] :  0x00000038
Mem[0x00000038] :  0xa023004c (10100000001000110000000001001100)
101000 | 00001 | 00011 | 0000000001001100
SB       rs: 1   rt: 3   imm:0x004c
R[ 1] :  0x80000001 (10000000000000000000000000000001)
R[ 3] :  0x0002a001 (00000000000000101010000000000001)
alu_a <= 0x80000001 (10000000000000000000000000000001)
alu_b <= 0x0000004c (00000000000000000000000001001100)
alu_r :  0x8000004d (10000000000000000000000001001101)
MMU[0x8000004d] :  0x0000004d
Mem[0x0000004d] <= 0x01 (00000001)

fetch instr @ 0x8000003c (10000000000000000000000000111100)
MMU[0x8000003c] :  0x0000003c
Mem[0x0000003c] :  0x3c028000 (00111100000000101000000000000000)
001111 | 00000 | 00010 | 1000000000000000
LUI      rs: 0   rt: 2   imm:0x8000
R[ 2] <= 0x80000000 (10000000000000000000000000000000)

fetch instr @ 0x80000040 (10000000000000000000000001000000)
MMU[0x80000040] :  0x00000040
Mem[0x00000040] :  0x8c42004c (10001100010000100000000001001100)
100011 | 00010 | 00010 | 0000000001001100
LW       rs: 2   rt: 2   imm:0x004c
R[ 2] :  0x80000000 (10000000000000000000000000000000)
alu_a <= 0x80000000 (10000000000000000000000000000000)
alu_b <= 0x0000004c (00000000000000000000000001001100)
alu_r :  0x8000004c (10000000000000000000000001001100)
MMU[0x8000004c] :  0x0000004c
Mem[0x0000004c] :  0xffff018e (11111111111111110000000110001110)
R[ 2] <= 0xffff018e (11111111111111110000000110001110)

fetch instr @ 0x80000044 (10000000000000000000000001000100)
MMU[0x80000044] :  0x00000044
Mem[0x00000044] :  0x0000000c (00000000000000000000000000001100)
000000 | 00000 | 00000 | 00000 | 00000 | 001100
SPECIAL  rs: 0   rt: 0   rd: 0   sa: 0   SYSCALL
halt
